src/PSR7/Validator.php

```diff
@@ -41,6 +51,15 @@ public static function fromYaml(string $yaml, ?CacheItemPoolInterface $cache = n
         return new static($oas);
     }
 
+    /**
+     * @param string $json
+     * @param CacheItemPoolInterface|null $cache
+     *
+     * @return static
```

Comment:
this and below changes would allow IDE to know that `ServerRequestValidator::fromYaml` returns `ServerRequestValidator`, not `Validator` (as `self`)

Reply:
Good point!